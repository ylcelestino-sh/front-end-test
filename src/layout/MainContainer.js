import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from "react-redux";
import { FiChevronLeft } from 'react-icons/fi';

import { useLocation, useNavigate } from 'react-router-dom';
import BadgeNotification from "../components/BadgeNotification";
import {
  Box,
  Flex,
  Stack,
  useColorModeValue,
  Container,
  Breadcrumb,
  BreadcrumbLink,
  BreadcrumbItem,
} from '@chakra-ui/react';

export const MainContainer = ({ children }) => {
  const navigate = useNavigate();
  const { pathname } = useLocation();
  const { countProductInCart } = useSelector((state) => state.product);
 
  return (
    <>
      <Box overflow="hidden" position="fixed" top="0" width="100%" zIndex="999">
        <Flex
          bg={useColorModeValue("white", "gray.800")}
          color={useColorModeValue("gray.600", "white")}
          minH={"60px"}
          py={{ base: 2 }}
          px={{ base: 4 }}
          borderBottom={1}
          borderStyle={"solid"}
          borderColor={useColorModeValue("gray.200", "gray.900")}
          align={"center"}
        >
          <Stack
            flex={{ base: 2, md: 1 }}
            justify={"flex-start"}
            direction={"row"}
            spacing={2}
          >
            <Breadcrumb separator={<FiChevronLeft color="gray.500" /> }>
              <BreadcrumbItem>
                <BreadcrumbLink onClick={() => navigate("/product")}>
                  Products
                </BreadcrumbLink>
              </BreadcrumbItem>
              {pathname.split("/")[1] === "product" && (
                <BreadcrumbItem isCurrentPage>
                  <BreadcrumbLink>Products Detail</BreadcrumbLink>
                </BreadcrumbItem>
              )}
            </Breadcrumb>
          </Stack>

          <BadgeNotification itemsNumbers={countProductInCart} />
        </Flex>
      </Box>
      <Container maxW="container.lg" pt={3}>
        {children}
      </Container>
    </>
  );
};

MainContainer.propTypes = {
  children: PropTypes.node,
};
export default MainContainer;
