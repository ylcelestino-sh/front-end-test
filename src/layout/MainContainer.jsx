import React from "react";
import PropTypes from "prop-types";
import { Container } from "@chakra-ui/react";
import Header from "./Header";

export const MainContainer = ({ children }) => {
  return (
    <>
      <Header />
      <Container maxW="container.lg" pt={3}  id="container">
        {children}
      </Container>
    </>
  );
};

MainContainer.propTypes = {
  children: PropTypes.node,
};

export default MainContainer;
