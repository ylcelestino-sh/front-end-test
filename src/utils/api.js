import axios from 'axios';

class ApiService {

  constructor() {
    this.baseUrl = process.env.REACT_APP_API_URL || 'https://front-test-api.herokuapp.com/';
    this.axiosInstance = axios.create({
      baseURL: this.baseUrl,
      headers: this.getHeaders(),
    });
  }
  // get Headers for create axios instance
  getHeaders() {
    const headers = {
      'content-type': 'application/json',
    };
    return headers;
  }
}
export default ApiService;