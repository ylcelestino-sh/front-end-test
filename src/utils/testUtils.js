import React from "react";
import PropTypes from "prop-types";
import { combineReducers, configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { setupStore } from "../app/store";
import { ChakraProvider } from "@chakra-ui/react";
import { BrowserRouter as Router } from "react-router-dom";

/*
 * Modified version of RTK Query's helper function:
 * https://github.com/reduxjs/redux-toolkit/blob/master/packages/toolkit/src/query/tests/helpers.tsx
 */

const AllProviders = ({ children }) => {
  AllProviders.propTypes = {
    children: PropTypes.node,
  };
  return (
    <Provider store={setupStore()}>
      <ChakraProvider>
        <Router>{children}</Router>
      </ChakraProvider>
    </Provider>
  );
};

export function withProvider(store) {
  return function Wrapper({ children }) {
    Wrapper.propTypes = {
      children: PropTypes.node,
    };
    return <Provider store={store}>{children}</Provider>
  }
}

const renderWithProviders = (
  ui,
  { route = "/products" } = {},
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  } = {}
) => {
  renderWithProviders.propTypes = {
    ui: PropTypes.node,
  };
  window.history.pushState({}, "Test page", route);

  return {
    store,
    AllProviders,
    ...render(ui, { wrapper: AllProviders, ...renderOptions }),
  };
};

const getStore = ({api, extraReducers}) => {
  return configureStore({
    reducer: combineReducers({
      [api.reducerPath]: api.reducer,
      ...extraReducers,
    }),
    middleware: (gdm) =>
      gdm({ serializableCheck: false, immutableCheck: false }).concat(
        api.middleware
      ),
  });
};

const setupApiStore = (api, extraReducers = {}) => {
  const initialStore = getStore({ api, extraReducers });
  const refObj = {
    api,
    store: initialStore,
    wrapper: withProvider({initialStore}),
  };

  const store = getStore({ api, extraReducers });
  refObj.store = store;

  return refObj;
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { renderWithProviders as render, setupApiStore };
