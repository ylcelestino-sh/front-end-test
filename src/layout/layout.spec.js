import React from "react";
import { render } from "../utils/testUtils";
import MainContainer from "./MainContainer";

test("Render main container", () => {
  const { container } = render(<MainContainer />);
 const header = container.querySelector('#header');
 const mainContainer = container.querySelector('#container');

  expect(mainContainer).toBeInTheDocument();
  expect(header).toBeInTheDocument();

  expect(mainContainer).toMatchSnapshot();
  expect(header).toMatchSnapshot();
    
});
