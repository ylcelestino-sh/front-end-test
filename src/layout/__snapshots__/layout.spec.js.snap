// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render main container 1`] = `
<div
  class="chakra-container css-pr22bl"
  id="container"
/>
`;

exports[`Render main container 2`] = `
<div
  class="css-rw02bh"
  id="header"
>
  <div
    class="css-1v5zkfs"
  >
    <div
      class="chakra-stack css-998b3j"
    >
      <nav
        aria-label="breadcrumb"
        class="chakra-breadcrumb css-0"
      >
        <ol
          class="chakra-breadcrumb__list css-70qvj9"
        >
          <li
            class="chakra-breadcrumb__list-item css-18biwo"
          >
            <a
              class="chakra-breadcrumb__link css-spn4bz"
            >
              Products
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <button
      aria-label="Notifications"
      class="chakra-button css-9ek20f"
      type="button"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="9"
          cy="21"
          r="1"
        />
        <circle
          cx="20"
          cy="21"
          r="1"
        />
        <path
          d="M1 1h4l2.68 13.39a2 2 0 0 0 2 1.61h9.72a2 2 0 0 0 2-1.61L23 6H6"
        />
      </svg>
      <span
        class="css-1szyxpv"
        data-testid="itemNumber"
        variant="ghost"
      >
        0
      </span>
    </button>
  </div>
</div>
`;
